third_party/move/move-compiler-v2/src/plan_builder.rs

```diff
@@ -0,0 +1,700 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::options::Options;
+use codespan_reporting::diagnostic::Severity;
+use move_command_line_common::{address::NumericalAddress, parser::NumberFormat};
+use move_compiler::{
+    shared::known_attributes::{AttributeKind, TestingAttribute},
+    unit_test::{ExpectedFailure, ExpectedMoveError, ModuleTestPlan, TestCase},
+};
+use move_core_types::{
+    identifier::Identifier, language_storage::ModuleId, value::MoveValue, vm_status::StatusCode,
+};
+use move_model::{
+    ast::{Address, Attribute, AttributeValue, ModuleName, Value},
+    model::{FunctionEnv, GlobalEnv, Loc, ModuleEnv, Parameter},
+    symbol::Symbol,
+    ty::{PrimitiveType, Type},
+};
+use num::{BigInt, ToPrimitive};
+use std::collections::BTreeMap;
+
+//***************************************************************************
+// Test Plan Building
+//***************************************************************************
+
+// Constructs a test plan for each module in `env.target`. This also validates the structure of the
+// attributes as the test plan is constructed.
+pub fn construct_test_plan(
+    env: &GlobalEnv,
+    package_filter: Option<Symbol>,
+) -> Option<Vec<ModuleTestPlan>> {
+    let options = env.get_extension::<Options>().expect("options");
+    if !options.compile_test_code {
+        return None;
+    }
+
+    Some(
+        env.get_modules()
+            .filter_map(|module| {
+                if module.is_target() {
+                    construct_module_test_plan(env, package_filter, module)
+                } else {
+                    None
+                }
+            })
+            .collect(),
+    )
+}
+
+fn construct_module_test_plan(
+    env: &GlobalEnv,
+    _package_filter: Option<Symbol>,
+    module: ModuleEnv,
+) -> Option<ModuleTestPlan> {
+    // BUGBUG: TODO: what is a package?
+    // if package_filter.is_some() && module.package_name != package_filter {
```

Comment:
I am assuming this is something that is being tackled soon (otherwise, we should ideally not have commented out code).